services-camel/src/main/java/edu/unc/lib/dl/services/camel/images/AddDerivativeProcessor.java

```diff
@@ -48,61 +45,41 @@
 
-    private final RepositoryObjectLoader repoObjLoader;
-    private final String slug;
     private final String fileExtension;
-    private final String mimetype;
-
-    private final int maxRetries;
-    private final long retryDelay;
+    private final String derivativeBasePath;
 
-    public AddDerivativeProcessor(RepositoryObjectLoader repoObjLoader, String slug, String fileExtension,
-            String mimetype, int maxRetries, long retryDelay) {
-        this.repoObjLoader = repoObjLoader;
-        this.slug = slug;
+    public AddDerivativeProcessor(String fileExtension, String derivativeBasePath) {
         this.fileExtension = fileExtension;
-        this.maxRetries = maxRetries;
-        this.retryDelay = retryDelay;
-        this.mimetype = mimetype;
+        this.derivativeBasePath = derivativeBasePath;
     }
 
     @Override
     public void process(Exchange exchange) throws Exception {
         Message in = exchange.getIn();
         String binaryUri = (String) in.getHeader(FCREPO_URI);
-        int retryAttempt = 0;
+        String binarySubPath = (String) in.getHeader(CdrBinarySubPath);
 
         final ExecResult result = (ExecResult) in.getBody();
 
         String derivativePath = new BufferedReader(new InputStreamReader(result.getStdout()))
                 .lines().collect(Collectors.joining("\n"));
 
-        while (true) {
-            try {
-                ingestFile(binaryUri, mimetype, derivativePath);
-                break;
-            } catch (Exception e) {
-                if (retryAttempt == maxRetries) {
-                    throw e;
-                }
-
-                retryAttempt++;
-                log.info("Unable to add derivative for {} from {}. Retrying, attempt {}",
-                        binaryUri, derivativePath, retryAttempt);
-                TimeUnit.MILLISECONDS.sleep(retryDelay);
-            }
-        }
+        moveFile(binaryUri, binarySubPath, derivativePath);
     }
 
-    private void ingestFile(String binaryUri, String binaryMimeType, String derivativePath)
-            throws FileNotFoundException {
-        String filename = derivativePath.substring(derivativePath.lastIndexOf('/') + 1);
-        InputStream binaryStream = new FileInputStream(derivativePath + "." + fileExtension);
+    private void moveFile(String binaryUri, String binarySubPath, String derivativeTmpPath)
+            throws IOException {
+        Path derivative_path = Paths.get(derivativeBasePath,  binarySubPath + "." + fileExtension);
+        File derivative = derivative_path.toFile();
+        File parentDir = derivative.getParentFile();
 
-        BinaryObject binary = repoObjLoader.getBinaryObject(PIDs.get(binaryUri));
-        FileObject parent = (FileObject) binary.getParent();
-        parent.addDerivative(slug, binaryStream, filename, binaryMimeType, PcdmUse.ThumbnailImage);
+        if (parentDir != null) {
+            parentDir.mkdirs();
+        }
+        derivative.createNewFile();
```

Comment:
Looks like this line isn't needed, the `move` command creates the destination file if it doesn't exist